index.html

```diff
@@ -975,6 +980,86 @@ <h2>Methods</h2>
             selected candidate pair =].
           </p>
         </dd>
+        <dt>
+          <dfn>removeCandidatePairs</dfn>
+        </dt>
+        <dd>
+          <p>
+            The {{removeCandidatePairs}} method removes the provided candidate pairs, making them unavailable to use
+            for transport. The [= ICE agent =] may free up candidates once all the associated candidate pairs have been removed.
+          </p>
+          <p>
+            When this method is invoked, the [= user agent =] MUST run the following steps:
+          </p>
+          <ol class="algorithm">
+            <li>
+              <p>
+                Let |connection:RTCPeerConnection| be the {{RTCPeerConnection}} object associated with this [= ICE agent =].
+              </p>
+            </li>
+            <li>
+              <p>
+                If <var>connection</var>.{{RTCPeerConnection/[[IsClosed]]}} is
+                <code>true</code>, [= exception/throw =] an
+                {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                Let |transport:RTCIceTransport| be the associated {{RTCIceTransport}} object.
+              </p>
+            </li>
+            <li>
+              <p>
+                If |transport|.{{RTCIceTransport/[[IceTransportState]]}} is either of
+                {{RTCIceTransportState/"new"}}, {{RTCIceTransportState/"failed"}} or {{RTCIceTransportState/"closed"}}, [=
+                exception/throw =]
+                an {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                If |transport|.{{RTCIceTransport/[[ProposalPending]]}} is <code>true</code>, [= exception/throw =] an
+                {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                Let |p:Promise| be a new promise.
+              </p>
+            </li>
+            <li>
+              <p>
+                In parallel, instruct the [= ICE agent =] to remove the candidate pairs indicated by |candidatePairs|, ignoring any unknown or invalid candidate pairs.
```

Comment:
What's the requirement on the ICE agent here? What does "remove" mean in the context of the ICE agent? Is it a ban against adding it again later?

Reply:
"Remove" stops connectivity checks and prevents the use of the candidate pair for sending data after removal. This is similar to the ICE agent discarding duplicate and lower priority pairs. The removed pairs cannot be added back again until candidates are gathered again, which is only possible today with an ICE restart.

Reply:
Can we link to prose in [RFC8445](https://www.rfc-editor.org/rfc/rfc8445) that allow user agents to modify its operation in this way?

Reply:
"Remove" isn't explicitly defined as a concept in RFC8445. It is something the ICE agent just does during the course of another operation:

1. [Section 6.1.2.4](https://www.rfc-editor.org/rfc/rfc8445#section-6.1.2.4) Pruning Pairs: The agent prunes each checklist.  This is done by **removing** a candidate pair if it is redundant with a higher-priority candidate pair in the same checklist.

2. [Section 6.1.2.5](https://www.rfc-editor.org/rfc/rfc8445#section-6.1.2.5) Removing Lower-Priority Pairs: The limit is enforced by, within in each checklist, **discarding** lower-priority candidate pairs...

So how about this as the requirement for the ICE agent:
1. Remove the provided candidate pair from all Checklists in which it is present.
2. [Update the states](https://www.rfc-editor.org/rfc/rfc8445#section-7.2.5.4) of the affected Checklists.
3. If any local candidates are now unused because they are no longer paired with any remote candidates, the ICE agent can [free those candidates](https://www.rfc-editor.org/rfc/rfc8445#section-8.3).

This means removed candidate pairs are not added back (at least not until regathering is possible).